ow_materials/include/MaterialIntegrator.h

```diff
@@ -49,15 +53,36 @@ class MaterialIntegrator
 
   ros::Publisher m_dug_points_pub;
 
+  // when enough time has passed since the last onModification call, this
+  // triggers logging of statistics about blend operation performance
+  ros::Timer m_stat_log_timeout;
+
   HandleBulkCallback m_handle_bulk_cb;
 
   ColorizerCallback m_colorizer_cb;
 
   AxisAlignedGrid<MaterialBlend> const *m_grid;
 
+  std::uint32_t m_next_expected_seq = 0u;
+
+  // used to compute performance statistics on batches of modifications
+  std::atomic<std::uint32_t> m_batch_total_modifications = 0u;
+  std::atomic<std::int64_t> m_batch_accumulated_processor_time = 0L;
+  std::atomic<std::uint32_t> m_batch_accumulated_merges = 0L;
+
+  // used to asynchronously parallelize calls to integrate
+  boost::asio::thread_pool m_threads;
+
+  void onStatLogTimeout(const ros::TimerEvent&);
+
+  void resetStatLogTimeout();
+
   void onModification(
     const ow_dynamic_terrain::modified_terrain_diff::ConstPtr &msg);
 
+  void integrate(
```

Comment:
It would help to have a comment with a colloquial explanation of what integrate does. The same is true for other methods, but integrate seems especially meaty.

Reply:
Added some more comments and clarified variable names